internal/core/application/service.go

```diff
@@ -765,21 +779,73 @@ func (s *service) FinalizeOffchainTx(
 		}
 	}()
 
-	finalCheckpointTxsMap := make(map[string]string)
+	decodedCheckpointTxs := make(map[string]*psbt.Packet)
 	for _, checkpoint := range finalCheckpointTxs {
 		// verify the tapscript signatures
-		valid, checkpointTxid, err := s.builder.VerifyTapscriptPartialSigs(checkpoint)
+		var valid bool
+		var checkpointTxid string
+		valid, checkpointTxid, err = s.builder.VerifyTapscriptPartialSigs(checkpoint)
```

Comment:
let's return the raw checkpoint tx here instead of just its txid since it seems we need it later? doesn't really make sense to deserialize the tx once again just above